cps/web.py

```diff
@@ -2719,6 +2719,35 @@ def edit_book(book_id):
 
     # Update book
     edited_books_id = set()
+
+    # Check and handle Uploaded file
+    if 'btn-upload-format' in request.files and '.' in request.files['btn-upload-format'].filename:
+        requested_file = request.files['btn-upload-format']
+        file_ext = requested_file.filename.rsplit('.', 1)[-1].lower()
+        if file_ext not in ALLOWED_EXTENSIONS:
+            flash(_('File extension "%s" is not allowed to be uploaded to this server' % file_ext), category="error")
+            return redirect(url_for('index'))
+
+        file_name = book.path.rsplit('/', 1)[-1]
+        filepath = config.config_calibre_dir + os.sep + book.path
+        filepath = os.path.normpath(filepath)
+        saved_filename = filepath + os.sep + file_name + '.' + file_ext
+        file_size = os.path.getsize(requested_file.stream.name)
```

Comment:
This line caused a problem for me. When I moved it to be right before line 2743 and changed it to `file_size = os.path.getsize(saved_filename)` it worked correctly.

Reply:
Cool, moved the line to prevent the error.